acceptance/tests/resource/exec/should_accept_large_output.rb

```diff
@@ -0,0 +1,33 @@
+test_name "tests that puppet correctly captures large and empty output."
+
+@testfilename = master.tmpfile('should_accept_large_output').split('/')[-1]
+
+agents.each do |agent|
+  testfile =
+    if /windows/ =~ agent[:platform]
+      "C:/#{@testfilename}"
+    else
+      "/tmp/#{@testfilename}"
+    end
+
+  # Generate >64KB file to exceed pipe buffer.
+  create_remote_file(agent, testfile, <<-EOF)
+Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
+aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
+occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
+EOF
+  on(agent, "for i in {1..10}; do cat #{testfile} #{testfile} > #{testfile}.2 && mv #{testfile}.2 #{testfile}; done")
```

Comment:
Why not just generate 64k+ content on the Ruby side? It will send more over the wire, but will remove the Bash-ism above. There will be a QA undertaking shortly to prune / rewrite tests where possible.  The less Bash shows up in tests, the better I think.